core/src/main/java/io/aiven/kafka/tieredstorage/manifest/SegmentIndexesV1Builder.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.manifest;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.kafka.server.log.remote.storage.RemoteStorageManager.IndexType;
+
+public class SegmentIndexesV1Builder {
+    final Map<IndexType, SegmentIndexV1> indexes = new HashMap<>(IndexType.values().length);
+    int currentPosition = 0;
+
+    public SegmentIndexesV1Builder add(final IndexType indexType, final int size) {
+        if (indexes.containsKey(indexType)) {
+            throw new IllegalStateException("Index " + indexType + " is already added");
+        }
+        switch (indexType) {
+            case OFFSET:
+                indexes.put(IndexType.OFFSET, new SegmentIndexV1(0, size));
+                currentPosition += size;
+                break;
+            case TIMESTAMP:
+                indexes.put(IndexType.TIMESTAMP, new SegmentIndexV1(currentPosition, size));
+                currentPosition += size;
+                break;
+            case PRODUCER_SNAPSHOT:
+                indexes.put(IndexType.PRODUCER_SNAPSHOT, new SegmentIndexV1(currentPosition, size));
+                currentPosition += size;
+                break;
+            case LEADER_EPOCH:
+                indexes.put(IndexType.LEADER_EPOCH, new SegmentIndexV1(currentPosition, size));
+                currentPosition += size;
+                break;
+            case TRANSACTION:
+                indexes.put(IndexType.TRANSACTION, new SegmentIndexV1(currentPosition, size));
+                currentPosition += size;
+                break;
+            default:
+                throw new IllegalStateException("Unknown index type");
+        }
```

Comment:
I think the whole switch could be replaced with
```suggestion
        indexes.put(indexType, new SegmentIndexV1(currentPosition, size));
        currentPosition += size;
```

Reply:
of course, 🤦🏽